sdk/storage/storage-blob/src/pollers/BlobStartCopyFromUrlPoller.ts

```diff
@@ -138,21 +153,22 @@ const cancel: BlobBeginCopyFromURLPollOperation["cancel"] = async function cance
   const state = this.state;
   const { copyId } = state;
   if (state.isCompleted) {
-    return makeBlobBeginCopyFromURLPollOperation(state);
+    return makeBlobBeginCopyFromURLPollOperation(state, this.lastResponse);
   }
 
   if (!copyId) {
     state.isCancelled = true;
-    return makeBlobBeginCopyFromURLPollOperation(state);
+    return makeBlobBeginCopyFromURLPollOperation(state, this.lastResponse);
   }
 
   // if abortCopyFromURL throws, it will bubble up to user's poller.cancelOperation call
-  await state.blobClient.abortCopyFromURL(copyId, {
+  const result = await state.blobClient.abortCopyFromURL(copyId, {
     abortSignal: options.abortSignal
   });
   state.isCancelled = true;
+  state.result = result;
 
-  return makeBlobBeginCopyFromURLPollOperation(state);
+  return makeBlobBeginCopyFromURLPollOperation(state, result);
```

Comment:
Currently we don't try-catch the errors from `startCopyFromURL` and `abortCopyFromURL`.  Should we wrap them with try-catch so we can return the results via `getLastResponse()`?

Reply:
From getLastResponse's semantic, it looks like it will always maintain last raw response get from network. try-catch and save the return result might help to keep the consistency. While not throwing might be a breaking change for customer, and handling error through throwing looks much nature. What about try-catch, save the result, and rethrow after save the result(please check if stack trace can be kept)? If the stack trace cannot be kept, it looks not try-catch would be better.

---
In reply to: [364920819](https://github.com/Azure/azure-sdk-for-js/pull/6561#discussion_r364920819) [](ancestors = 364920819)

Reply:
catch and re-throw sounds good. Although currently we either return a PollOperation, or throw from the operations. I store the last response on the returned PollOperation object so in the error case it will not be available.

@bterlson @chradek @xirzec any thoughts?

Reply:
I think I'm a little confused here, since in the throw case we don't have a new result, right? Unless that is part of the error object?

Reply:
`poller.getLastResponse()` returns either the raw response, or `state.error`.  In some of the poll operations (e.g., `getProperties` to check progress) we handle the error from service and set `state.error`. but for `startCopyFromURL` and `abortCopyFromURL` we let the error bubble to users.